consent-ui/src/app/ais/sca-select-page/sca-select-page.component.ts

```diff
@@ -1,48 +1,74 @@
 
 @Component({
-  selector: 'consent-app-sca-select-page',
-  templateUrl: './sca-select-page.component.html',
-  styleUrls: ['./sca-select-page.component.scss']
+    selector: 'consent-app-sca-select-page',
+    templateUrl: './sca-select-page.component.html',
+    styleUrls: ['./sca-select-page.component.scss']
 })
 export class ScaSelectPageComponent implements OnInit {
-  private redirectTo: string;
-  private authorizationId: string;
-  private aisConsent: AisConsentToGrant;
 
-  constructor(
-    private activatedRoute: ActivatedRoute,
-    private sessionService: SessionService,
-    private consentAuthorisation: ConsentAuthorizationService
-  ) {
-  }
+    authorizationSessionId: string = '';
+    redirectCode: string = '';
+    scaMethodForm: FormGroup;
+    scaMethods: ScaUserData[];
+    private subscriptions: Subscription[] = [];
+
+    constructor(private sessionService: SessionService,
+                private consentAuthorizationService: ConsentAuthorizationService,
+                private route: ActivatedRoute,
+                private formBuilder: FormBuilder) {
+    }
+
+    ngOnInit() {
+        this.initialScaMethodForm();
+        this.authorizationSessionId = this.route.parent.snapshot.paramMap.get("authId");
+        this.redirectCode = this.sessionService.getRedirectCode(this.authorizationSessionId);
+        this.loadAvailableMethods();
+    }
 
-  ngOnInit() {
-    this.activatedRoute.parent.params.subscribe(res => {
-      this.authorizationId = res.authId;
-      this.aisConsent = ConsentUtil.getOrDefault(this.authorizationId, this.sessionService);
-      this.loadRedirectUri();
-    });
-  }
+    onSubmit(): void {
+        this.subscriptions.push(
+            this.consentAuthorizationService.embeddedUsingPOST(
+                this.authorizationSessionId,
+                StubUtil.X_REQUEST_ID, // TODO: real values instead of stubs
+                StubUtil.X_XSRF_TOKEN, // TODO: real values instead of stubs
+                this.redirectCode,
+                {scaAuthenticationData: {SCA_CHALLENGE_ID: this.scaMethodForm.get('selectedMethodValue').value}},
+                "response"
+            ).subscribe(
+                res => {
+                    // redirect to the provided location
+                    console.log("REDIRECTING TO: " + res.headers.get(ApiHeaders.LOCATION));
+                    this.sessionService.setRedirectCode(this.authorizationSessionId, res.headers.get(ApiHeaders.REDIRECT_CODE));
+                    window.location.href = res.headers.get(ApiHeaders.LOCATION);
+                },
+                error => {
+                    console.log(error);
+                    // window.location.href = error.url;
+                })
+        );
+    }
 
-  onConfirm() {
-    window.location.href = this.redirectTo;
-  }
+    private loadAvailableMethods(): void {
+        this.consentAuthorizationService.authUsingGET(this.authorizationSessionId, this.redirectCode, 'response')
```

Comment:
If you want to keep unsubsciption logic, maybe it makes sense to add this one to subscriptions too